src/engine/Server.cpp

```diff
@@ -751,6 +753,19 @@ MediaType Server::determineMediaType(
   return mediaType.value();
 }
 
+// ____________________________________________________________________________
+std::pair<std::shared_ptr<ad_utility::websocket::QueryHub>,
+          ad_utility::websocket::MessageSender>
+Server::createMessageSender(auto& queryHub_, const auto& request,
+                            const string& operation) {
+  auto queryHub = queryHub_.lock();
+  AD_CORRECTNESS_CHECK(queryHub);
+  ad_utility::websocket::MessageSender messageSender{
+      getQueryId(request, operation), *queryHub};
+  // TODO<qup42> is it required to keep the queryHub alive?
```

Comment:
I just checked, We have gone through greath lengths to even make the case work where the queryHub is destroyed to early (which anyway cannot happen here, but only in unit tests).
So you can simply return the `MessageSender` and all calls etc.  become simpler.